lib/puppet/pops/types/type_asserter.rb

```diff
@@ -1,6 +1,7 @@
 # Utility module for type assertion
 #
-module Puppet::Pops::Types::TypeAsserter
+module Puppet::Pops::Types
+module TypeAsserter
```

Comment:
It's Ruby. Never ceases to surprise you :-) This piece of code can be used to verify:
```
module A
  class B
  end
end

module A
  module C
    class D
      def initialize
        B.new
      end
    end
  end
end

module A::C
  class E
    def initialize
      B.new
    end
  end
end

A::C::D.new # succeeds
A::C::E.new # fails with "uninitialized constant A::C::E::B (NameError)"
```